lib/active_merchant/billing/gateways/opp.rb

```diff
@@ -0,0 +1,399 @@
+    #   )
+    #
+    #   # set up credit card object as in main ActiveMerchant example
+    #   creditcard = ActiveMerchant::Billing::CreditCard.new(
+   #     :type       => 'visa',
+    #     :number     => '4242424242424242',
+    #     :month      => 8,
+    #     :year       => 2009,
+    #     :first_name => 'Bob',
+    #     :last_name  => 'Bobsen'
+    #     :verification_value: '123')
+    #
+    #   # Request: complete example, including address, billing address, shipping address, shopping cart
+    #    complete_request_options = {
+    #      order_id: "your merchant/shop order id", # alternative is to set merchantInvoiceId 
+    #      merchantTransactionId: "your merchant/shop transaction id",
+    #      address: address,
+    #      description: 'Store Purchase - Books',
+    #      riskWorkflow: false,
+    #      testMode: 'EXTERNAL' # or 'INTERNAL', valid only for test system
+    #      createRegistration: false, # payment details will be stored on the server an latter can be referenced
+    #
+    #     billing_address: {
+    #        address1: '123 Test Street',
+    #        city:     'Test',
+    #        state:    'TE',
+    #        zip:      'AB12CD',
+    #        country:  'GB',
+    #      },
+    #      shipping_address: {
+    #        name:     'Muton DeMicelis',
+    #        address1: 'My Street On Upiter, Apt 3.14/2.78',
+    #        city:     'Munich',
+    #        state:    'Bov',
+    #        zip:      '81675',
+    #        country:  'DE',
+    #      },
+    #      customer: {
+    #        merchantCustomerId:  "your merchant/customer id",
+    #        givenName:  'Jane',
+    #        surname:  'Jones',
+    #        birthDate:  '1965-05-01',
+    #        phone:  '(?!?)555-5555',
+    #        mobile:  '(?!?)234-23423',
+    #        email:  'jane@jones.com',
+    #        companyName:  'JJ Ltd.',
+    #        identificationDocType:  'PASSPORT',
+    #        identificationDocId:  'FakeID2342431234123',
+    #        ip:  101.102.103.104,
+    #      },
+    #      cart: {
+    #        items: [
+    #            { name: ' Bestseller Book', merchantItemId: 'isbn-0123456789012345', 
+    #              quantity: 1, type: 'book', price: 1.95, currency: 'EUR', description: 'Some item description',
+    #              tax: 7.0, shipping: 3.25, discount: 5.0
+    #            },                    
+    #            { name: 'Book 2', merchantItemId: 'isbn-0123456789012345', 
+    #              quantity: 1, type: 'book', price: 2.45, currency: 'EUR', description: 'Other item description',
+    #              tax: 7.0, shipping: 3.25, discount: 10.0
+    #            }                    
+    #        ],
+    #      },
+    #    }
+    #    
+    #    # Request: minimal example
+    #    minimal_request_options = {
+    #      order_id: "your merchant/shop order id", # alternative is to set merchantInvoiceId 
+    #      merchantTransactionId: "your merchant/shop transaction id",
```

Comment:
ActiveMerchant doesn't allow non-standard options to be required, so for merging `merchant_transaction_id` must be made optional and/or mapped to a standard option like `order_id`.